proposals/NNNN-implicit-self-explicit-capture.md

```diff
@@ -0,0 +1,189 @@
+# Increase availability of implicit `self` in `@escaping` closures when reference cycles are unlikely to occur
+
+* Proposal: [SE-NNNN](NNNN-implicit-self-explicit-capture.md)
+* Authors: [Frederick Kellison-Linn](https://github.com/jumhyn)
+* Review Manager: TBD
+* Status: **Awaiting Review**
+* Implementation: [apple/swift#23934](https://github.com/apple/swift/pull/23934/)
+* Bug: [SR-10218](https://bugs.swift.org/browse/SR-10218)
+* Forum threads: [Discussion](https://forums.swift.org/t/review-capture-semantics-of-self/22017), [Pitch](https://forums.swift.org/t/allow-implicit-self-in-escaping-closures-when-self-is-explicitly-captured/22590)
+
+## Introduction
+
+Modify the rule that all uses of `self` in escaping closures must be explicit by allowing for implicit uses of `self` in situations where the user has already made their intent explicit, or where strong reference cycles are otherwise unlikely to occur. There are two situations covered by this proposal. The first is when the user has explicitly captured `self` in the closures capture list, so that the following would compile without error:
```

Comment:
```suggestion
Modify the rule that all uses of `self` in escaping closures must be explicit by allowing for implicit uses of `self` in situations where the user has already made their intent explicit, or where strong reference cycles are otherwise unlikely to occur. There are two situations covered by this proposal. The first is when the user has explicitly captured `self` in the closure's capture list, so that the following would compile without error:
```

Reply:
Markdown with manual line breaks is more source-control-friendly, FWIW.